### PR TITLE
Create comments model

### LINKED ISSUE #8: Enable the backend to add comments to posts

### PR DESCRIPTION
**Pull request by r-coh**
Fixes #8

## Proposed Changes

  - add comments table
  - add tests for the same
  -

### CODE REVIEW

File: priv/repo/migrations/20180809102052_create_comments.exs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+defmodule QbBackend.Repo.Migrations.CreateComments do
+  use Ecto.Migration
+
+  def change do
+    create table(:comments, primary_key: false) do
+      add :id, :binary_id, primary_key: true
+      add :body, :string
+      #add :manual_id, references(:manuals, type: :binary_id)
```
Comment (zacck-zz) [inline priv/repo/migrations/20180809102052_create_comments.exs]: You can remove this since we will make a new migration to associate comments to manuals

**Review by zacck-zz — CHANGES_REQUESTED** [review]
Hey! this looks good!!

**Review by zacck-zz — APPROVED** [review]

**Review by zacck-zz — CHANGES_REQUESTED** [review]
add tests for the manuals

File: test/support/factory.ex [hunk 2]
```diff
@@ -31,7 +31,7 @@ defmodule QbBackend.Factory do
   def comment_factory do
     %Comment{
       body: sequence(:body, &"body-#{&1}"),
-      profile: build(:profile)
+      profile: build(:profile),
```
Comment (zacck-zz) [inline test/support/factory.ex]: is this last comma needed?

File: lib/qb_backend/posts/comment.ex [hunk 3]
```diff
@@ -0,0 +1,47 @@
+defmodule QbBackend.Posts.Comment do
```
Comment (zacck-zz) [inline lib/qb_backend/posts/comment.ex:1]: please run mix format for all these changes

File: lib/qb_backend/posts/posts.ex [hunk 4]
```diff
@@ -44,4 +44,35 @@ defmodule QbBackend.Posts do
     |> Repo.update()
   end
 
+  #Comments related functions
```
Comment (zacck-zz) [inline lib/qb_backend/posts/posts.ex]: Is this comment needed ?

File: test/lib/qb_backend/posts/posts_test.exs [hunk 5]
```diff
@@ -56,5 +58,40 @@ defmodule QbBackend.PostsTest do
         assert saved_manual.id == manual.id
         assert saved_manual.title == manual.title
     end
+
+  # Comment functions tests under Posts context
```
Comment (zacck-zz) [inline test/lib/qb_backend/posts/posts_test.exs]: this comment is unneccessary as the code is very clear

File: test/support/factory.ex [hunk 6]
```diff
@@ -24,6 +25,13 @@ defmodule QbBackend.Factory do
       role: sequence(:role, ["reader", "author", "publisher"]),
       avatar_link: sequence(:avatar_link, &"link-#{&1}"),
       user: build(:user)
+     }
+  end
+
+  def comment_factory do
+    %Comment{
+      body: sequence(:body, &"body-#{&1}"),
+      profile: build(:profile),
```
Comment (zacck-zz) [inline test/support/factory.ex]: is this last comma needed?

**Review by zacck-zz — CHANGES_REQUESTED** [review]
@br4inii  @WanjikuMac  such awesome teamwork! just a little push to go over the hill

File: test/support/factory.ex (hunk 2 above)
Comment (r-coh) [reply test/support/factory.ex]: not really,

**Review by zacck-zz — CHANGES_REQUESTED** [review]
Awesome work team @br4inii  @WanjikuMac

**Review by zacck-zz — APPROVED** [review]

**Pull request merged**